src/emqx_mgmt.erl

```diff
@@ -563,6 +566,44 @@ list_listeners(Node) when Node =:= node() ->
 list_listeners(Node) ->
     rpc_call(Node, list_listeners, [Node]).
 
+restart_listener(Proto, ListenOn) ->
+    ListenOn1 = case string:tokens(ListenOn, ":") of
+        [Port]     -> list_to_integer(Port);
+        [IP, Port] -> {IP, list_to_integer(Port)}
```

Comment:
If your listener.ssl.external is configured as `127.0.0.1:8883` then the comparison on line 600 will not work.

The `IP` should be converted to a tuple.  
e.g:
```erl
case inet:parse_address(IP) of
    {ok, IP1} -> {IP1, list_to_integer(Port)};
    {error,  Error} -> 
      %% Make an error return.
      ...
end
```

Reply:
In addition to testing 8883, we should also make sure that this command works well on other listens. e.g:
```
. /bin/emqx_ctl listeners restart mqtt:tcp 0.0.0.0:1883
. /bin/emqx_ctl listeners restart mqtt:tcp 127.0.0.1:11883
. /bin/emqx_ctl listeners restart mqtt:wss 8084
```

Reply:
Ok, I will add the conversion and test it with an IP configured, thanks for providing the code snippet!

I have tested with all listeners, they don't return error on restart, by I couldn't actually test e.g. ws *connections* re-connecting because I don't have that scenario. But I think I can try with the dashboard ws-test feature. At least the ws feature, but I will also try with the wss + certs scenario.